service/pixelated/resources/account_recovery_resource.py

```diff
@@ -52,10 +61,24 @@ def success_response(response):
             request.setResponseCode(OK)
             request.finish()
 
-        def error_response(response):
+        def error_response(failure):
+            log.warn(failure)
             request.setResponseCode(INTERNAL_SERVER_ERROR)
             request.finish()
 
-        d = defer.succeed('Done!')
+        d = self._validate_password(request)
         d.addCallbacks(success_response, error_response)
         return NOT_DONE_YET
+
+    def _get_post_form(self, request):
+        return json.loads(request.content.getvalue())
+
+    def _validate_password(self, request):
+        form = self._get_post_form(request)
+        password = form.get('password')
+        confirmPassword = form.get('confirmPassword')
+
+        if password == confirmPassword and len(password) >= 8 and len(password) <= 9999:
```

Comment:
I think this if should be the only thing inside a `validate_password` method, in my opinion. Everything else inside this method is treatment for a POST request. I would leave it in the POST method. Might even mean that `valid_password` return a boolean. Following a bit more of a single responsibility principle.